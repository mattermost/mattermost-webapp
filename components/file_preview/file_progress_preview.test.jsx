// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import FileProgressPreview from './file_progress_preview.jsx';

describe('component/file_preview/file_progress_preview', () => {
    const handleRemove = jest.fn();

    const fileInfo = {
        name: 'file',
        percent: 50,
        type: 'image/png',
    };

    const baseProps = {
        clientId: 'clientId',
        fileInfo,
        handleRemove,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <FileProgressPreview {...baseProps}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
