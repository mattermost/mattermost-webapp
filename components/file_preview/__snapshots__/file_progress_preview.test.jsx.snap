// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/file_preview/file_progress_preview should match snapshot 1`] = `
<div
  className="file-preview post-image__column"
  data-client-id="clientId"
  key="clientId"
>
  <div
    className="post-image__thumbnail"
  >
    <div
      className="file-icon image"
    />
  </div>
  <div
    className="post-image__details"
  >
    <div
      className="post-image__detail_wrapper"
    >
      <div
        className="post-image__detail"
      >
        <FilenameOverlay
          canDownload={false}
          compactDisplay={false}
          fileInfo={
            Object {
              "name": "file",
              "percent": 50,
              "type": "image/png",
            }
          }
          handleImageClick={null}
          index="clientId"
        />
        <span
          className="post-image__uploadingTxt"
        >
          <FormattedMessage
            defaultMessage="Uploading..."
            id="admin.plugin.uploading"
            values={Object {}}
          />
          <span>
             (50%)
          </span>
        </span>
        <ProgressBar
          active={false}
          bsClass="progress-bar"
          className="post-image__progressBar"
          isChild={false}
          max={100}
          min={0}
          now={50}
          srOnly={false}
          striped={false}
        />
      </div>
    </div>
    <div>
      <a
        className="file-preview__remove"
        onClick={[Function]}
      >
        <i
          className="fa fa-remove"
          title="Remove Icon"
        />
      </a>
    </div>
  </div>
</div>
`;
